compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs

```diff
@@ -825,14 +825,39 @@ renderValue world name = \case
         renderField (Field label mbValue) =
             renderValue world (name ++ [TL.toStrict label]) (fromJust mbValue)
 
+templateConName :: Identifier -> LF.Qualified LF.TypeConName
+templateConName (Identifier mbPkgId (TL.toStrict -> qualName)) = LF.Qualified pkgRef  mdN tpl
+  where (mdN, tpl) = case T.splitOn ":" qualName of
+          [modName, defN] -> (LF.ModuleName (T.splitOn "." modName) , LF.TypeConName (T.splitOn "." defN) )
+          _ -> error "malformed identifier"
+        pkgRef = case mbPkgId of
+                  Just (PackageIdentifier (Just (PackageIdentifierSumPackageId pkgId))) -> LF.PRImport $ LF.PackageId $ TL.toStrict pkgId
+                  Just (PackageIdentifier (Just (PackageIdentifierSumSelf _))) -> LF.PRSelf
+                  Just (PackageIdentifier Nothing) -> error "malformed identifier"
+                  Nothing -> LF.PRSelf
+
+typeConFieldsNames :: LF.World -> (LF.FieldName, LF.Type) -> [T.Text]
+typeConFieldsNames world (LF.FieldName fName, LF.TCon tcn) = map (TE.append (TE.append fName ".")) (templateConFields tcn world)
+typeConFieldsNames _ (LF.FieldName fName, _) = [fName]
+
+templateConFields :: LF.Qualified LF.TypeConName -> LF.World -> [T.Text]
+templateConFields qName world = case LF.lookupDataType qName world of
+    Right (LF.DefDataType _ _ _ _ (LF.DataRecord re) ) -> concatMap (typeConFieldsNames world) re
+    Right (LF.DefDataType _ _ _ _ (LF.DataVariant re) ) -> map (LF.unVariantConName . fst) re
```

Comment:
A variant type is something like `Either`. It should only produce one entry in the header, most likely `[""]`. I suggest you try to render something involving `Either` and it will become clear what the problem is.

Reply:
Instead of repeating this long pattern twice you could do something like
```haskell
case LF.lookupDataType qName world of
  Right dataType -> case LF.dataCons dataType of
    LF.DataRecord re -> ...
    LF.DataVariant re -> ...
```